pkg/master/master_test.go

```diff
@@ -73,3 +82,282 @@ func TestFindExternalAddress(t *testing.T) {
 		t.Errorf("expected findExternalAddress to fail on a node with missing ip information")
 	}
 }
+
+type Foo struct {
+	api.TypeMeta   `json:",inline"`
+	api.ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	SomeField  string `json:"someField"`
+	OtherField int    `json:"otherField"`
+}
+
+type FooList struct {
+	api.TypeMeta `json:",inline"`
+	api.ListMeta `json:"metadata,omitempty" description:"standard list metadata; see http://docs.k8s.io/api-conventions.md#metadata"`
+
+	items []Foo `json:"items"`
+}
+
+func initThirdParty(t *testing.T) (*tools.FakeEtcdClient, *httptest.Server) {
+	master := &Master{}
+	api := &expapi.ThirdPartyResource{
+		ObjectMeta: api.ObjectMeta{
+			Name: "foo.company.com",
+		},
+		Versions: []expapi.APIVersion{
+			{
+				APIGroup: "group",
+				Name:     "v1",
```

Comment:
If it's changed to "v3", the test will fail with:
master.go:794] Unable to setup thirdparty api: no kind "ThirdPartyResourceData" is registered for version "v3"